### PR TITLE
Handle fees

### LINKED ISSUE #39: Support fees
Starting with v0.11.0, celestia-appd require fees to be set for transactions (including PayForData).
Changes are already implemented in celestia-node v0.6.1. We need to implement them in go-cnc.

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

Resolves #39.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -44,10 +44,11 @@ func (c *Client) SubmitTx(ctx context.Context, tx []byte) /* TxResponse */ error
 	return errors.New("method SubmitTx not implemented")
 }
 
-func (c *Client) SubmitPFD(ctx context.Context, namespaceID [8]byte, data []byte, gasLimit uint64) (*TxResponse, error) {
+func (c *Client) SubmitPFD(ctx context.Context, namespaceID [8]byte, data []byte, fee int64, gasLimit uint64) (*TxResponse, error) {
 	req := SubmitPFDRequest{
 		NamespaceID: hex.EncodeToString(namespaceID[:]),
 		Data:        hex.EncodeToString(data),
+		Fee:         fee,
```
Comment (gupadhyaya) [inline client.go:51]: Since `GasPrice` is generally used along with `GasLimit`, is migration to `Fee` on purpose?

File: types.go [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-import "github.com/gogo/protobuf/types"
+import (
+	"github.com/gogo/protobuf/types"
+)
 
 // SubmitPFDRequest represents a request to submit a PayForData transaction.
 type SubmitPFDRequest struct {
 	NamespaceID string `json:"namespace_id"`
 	Data        string `json:"data"`
+	Fee         int64  `json:"fee"`
```
Comment (gupadhyaya) [inline types.go:11]: What is the expected range of values? With decimals in mind, would `int64` be sufficient? `GasPrice` is generally handled as big integer.

File: client.go (hunk 1 above)
Comment (tzdybal) [reply client.go:51]: go-cnc is 1:1 mapping for gateway API exposed by celestia node. This changes mimics their code. See: https://github.com/celestiaorg/celestia-node/commit/4734aac019e5d786be1148f8cbd0bc84ebc392a9

File: types.go (hunk 2 above)
Comment (tzdybal) [reply types.go:11]: Again, we my want to change this later, and we need would need to fix this also in celestia-node (see: previous comment).

**Review by gupadhyaya — APPROVED** [review]

File: types.go (hunk 2 above)
Comment (distractedm1nd) [reply types.go:11]: Fee has the same type as the `amount` field in other requests, as it ends up as a cosmos `math.Int`. Yes we might want to change this later

The `GasLimit` may also change to an int64 soon:
https://github.com/celestiaorg/celestia-app/issues/1133

But will keep go-cnc in mind when these changes occur!

**Review by distractedm1nd — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]
@distractedm1nd proxy approve

**Pull request merged**